pkg/installer/uninstall.go

```diff
@@ -225,25 +237,21 @@ func (in *Installer) kustomizeAndDelete(dir, file string) error {
 
 // gracefullyDeleteNS deletes a k8s namespace only once there are no pod running in said namespace,
 // then waits for the namespace to be removed from the cluster before returning no error
-func (in *Installer) gracefullyDeleteNS(namespaceManifest string) error {
-	namespaceName, err := pluginutils.GetFieldInManifest(namespaceManifest, "metadata", "name")
-	if err != nil {
-		return err
-	}
-	err = pluginutils.WaitFor(func() error {
-		return pluginutils.NoPodsInNS(in.clientConfig, namespaceName)
-	}, 120, 5)
+func (in *Installer) gracefullyDeleteNS(namespacename string) error {
+	err := pluginutils.WaitFor(func() error {
+		return pluginutils.NoResourcesInNS(in.clientConfig, namespacename)
+	}, 180, 10)
 	if err != nil {
 		return err
 	}
 
-	err = in.kubectlClient.Delete(context.TODO(), "", namespaceManifest, true)
+	err = pluginutils.DeleteNamespace(in.clientConfig, namespacename)
```

Comment:
nit: I'd use (here and elsewhere):
```
if err := pluginutils.DeleteNamespace(in.clientConfig, namespacename); err != nil {
  return err
}
```

Reply:
I don't really understand. Other places we use resource manifest to delete, this is the only place where we delete by name.

Reply:
It was just code style, when a function only returns an error you'd typically scope it to the if statement that checks it.  Not a big deal, I see it's done this way in most other places in the project.

Reply:
The project doesn't follow this pattern. I will fix this one, because i also prefer.

Reply:
yeah, I hadn't realised at the time that the project doesn't follow the pattern.  Not a big big deal, and better to be consistent.  Maybe we revisit later.